src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -234,29 +277,30 @@ public Result handle(
 
     @Override
     public ActivityTaskHandler.Result execute(ActivityInfoImpl info, Scope metricsScope) {
-      ActivityExecutionContext context = new LocalActivityExecutionContextImpl(info);
-      CurrentActivityExecutionContext.set(context);
+      ActivityExecutionContext context = new LocalActivityExecutionContextImpl(info, metricsScope);
       Optional<Payloads> input = info.getInput();
+      ActivityInboundCallsInterceptor inboundCallsInterceptor =
+          new POJOActivityInboundCallsInterceptor(activity, method);
+      for (ActivityInterceptor interceptor : interceptors) {
+        inboundCallsInterceptor = interceptor.interceptActivity(inboundCallsInterceptor);
+      }
+      inboundCallsInterceptor.init(context);
       try {
         Object[] args =
             dataConverter.arrayFromPayloads(
                 input, method.getParameterTypes(), method.getGenericParameterTypes());
-        Object result = method.invoke(activity, args);
+        Object result = inboundCallsInterceptor.execute(args);
```

Comment:
Minor nit - Looks like stack trace in this test JSON may be out of sync after this change:

https://github.com/temporalio/temporal-java-sdk/blob/master/src/test/resources/testAsyncActivityRetryHistory.json#L122